DependencyInjection/Compiler/SerializerConfigurationPass.php

```diff
@@ -33,13 +33,26 @@ public function process(ContainerBuilder $container)
         }
 
         if ($container->has('jms_serializer.serializer')) {
-            $container->setAlias('fos_rest.serializer', 'jms_serializer.serializer');
+            $container->setAlias('fos_rest.serializer', 'fos_rest.serializer.jms');
         } else {
             $container->removeDefinition('fos_rest.serializer.exception_wrapper_serialize_handler');
         }
 
         if ($container->has('serializer')) {
-            $container->setAlias('fos_rest.serializer', 'serializer');
+            $serializer = $container->findDefinition('serializer');
+            $class = $serializer->getClass();
+
+            if ('%' === $class[0]) {
+                $class = $container->getParameter(substr($class, 1, -1));
+            }
+
+            if (is_subclass_of($class, 'Symfony\Component\Serializer\SerializerInterface')) {
+                $container->setAlias('fos_rest.serializer', 'fos_rest.serializer.symfony');
+            } elseif (is_subclass_of($class, 'JMS\Serializer\SerializerInterface')) {
+                $container->setAlias('fos_rest.serializer', 'fos_rest.serializer.jms');
+            } else {
+                throw new \LogicException(sprintf('Cannot handle serializer of type "%s"', $class));
```

Comment:
I agree and that's what I mean with the early exit in the compiler pass when the `fos_rest.serializer` service is already set (which we may ease later by adding a config option for it). Or do you mean something else?